src/components/RegisterPage/RegisterPage.tsx

```diff
@@ -140,7 +140,11 @@ const RegisterPage = ({ registering, isRegistrationSuccess }) => {
                 )}
               </div>
 
-              <button className={styles.submitBtn} onClick={() => onSubmit}>
+              <button
+                className={styles.submitBtn}
+                type="submit"
+                onClick={() => onSubmit}
```

Comment:
have you tested this button is working? looks like you write function name onSubmit without calling it `onSubmit(...)`